### ISSUE TITLE
Focussing a monitor changes active desktop

### ISSUE THREAD

**Issue by Shrvi** [baskerville/bspwm#773]
### General Information
- Bspwm version: `0.9.3-49-g0d0fb94` from [bspwm-git (AUR)](https://aur.archlinux.org/packages/bspwm-git/)

### Problem description
I noticed two cases, there may be more, where `bspc monitor -f …` also changes the focused desktop on the focussed monitor:

1. When _activating_ a desktop on an unfocused monitor. Focussing the monitor later, focusses the monitor but also focusses the desktop beeing active before. See [Reproduction steps](#reproduction-steps).

2. When swapping the focussed desktop with an inactive desktop of an unfocussed monitor.
Focussing the monitor later, also focusses the desktop that was swapped to that monitor.

#### Minimal bspwmrc:
```sh
xrandr --output A --auto
xrandr --output B --auto

bspc monitor A -d 1 2
bspc monitor B -d 3 4
```

### Reproduction steps
1. bspc desktop -a 4
2. bspc monitor -f B

```
Before 1: WMA:O1:f2:LT:TT:G:mB:F3:f4:LT
After 1:  WMA:O1:f2:LT:TT:G:mB:f3:F4:LT
After 2:  WmA:O1:f2:LT:TT:G:MB:F3:f4:LT
Expected: WmA:O1:f2:LT:TT:G:MB:f3:F4:LT
```
We see, `monitor -f B` focuses monitor B and desktop `3`, but expected desktop `4`.

**Comment by baskerville** [comment]
Problem *1* should be fixed by 181e679.
Problem *2* is caused by the way we rewrite the history when objects are swapped. A fix is being prepared.

**Comment by baskerville** [comment]
The last problem should be fixed by 30e26a6.

**Comment by Shrvi** [comment]
It works.
Thank you for always being so fast :)
